src/main/java/com/disi/TravelPoints/service/UserService.java

```diff
@@ -38,9 +40,13 @@ public UserDetails authenticate(AuthenticateRequest request) throws Exception {
                 .build();
     }
 
-    public Long register(RegisterRequest request) {
-        var touristRole = roleService.findByRoleName(RoleName.TOURIST);
+    public Long register(RegisterRequest request) throws Exception {
+        final Optional<User> userByEmail = userRepository.findByEmail(request.getEmail());
+        if (userByEmail.isPresent()) {
+            throw new Exception("This email is already taken!");
+        }
 
+        var touristRole = roleService.findByRoleName(RoleName.TOURIST);
```

Comment:
Wasn't it enough to directly set the "tourist" role? Because each new registration will actually be a tourist.

Reply:
A User has a Role, not a RoleName.

Reply:
Ah, my bad, sorry.